### PR TITLE
Showmaker performance improvements (monthly)

### PR DESCRIPTION
**Pull request by thelovebug**
Refactoring the monthly showmaker to speed up the build process.
Changes to daily and weekly process as a result of the change of trackTrimSilence.

### CODE REVIEW

File: CLI/library.php [hunk 1]
```diff
@@ -308,6 +308,55 @@ function concatenateTracks($first, $second, $output, $remove_sources = true)
...
+ * @return boolean Success or Failure
+ */
+function concatenateMultiTracks($first, $last, $intermediate, $num_tracks, $output, $remove_sources = true)
+{
+
+//    print ('first: ' . $first . ' : ');
+//    print ('last: ' . $last . ' : ' );
+//    print ('intermediate: ' . $intermediate . ' : ');
+//    print ('num_tracks: ' . $num_tracks . ' : ');
+//    print ('output: ' . $output );
```
Comment (JonTheNiceGuy) [inline CLI/library.php]: Let's remove these debugging lines, as they're not really needed any more?

File: CLI/library.php [hunk 2]
```diff
@@ -308,6 +308,55 @@ function concatenateTracks($first, $second, $output, $remove_sources = true)
...
+
+	$multi = '';
+
+	for($counter=1;$counter<=$num_tracks;$counter++) {
+            $multi .=  '"' . str_replace( '{n}', $counter, $intermediate ) . '" ';
+        }
+
+	$cmd = str_replace( '{multi}', $multi, $cmd );
+
+//	print($cmd);
```
Comment (JonTheNiceGuy) [inline CLI/library.php]: And this one?

File: CLI/showmaker.php [hunk 3]
```diff
@@ -169,8 +169,10 @@
         if ( ! trackTrimSilence(Configuration::getWorkingDir() . '/' . $show_data['arrTracks'][1]['fileSource'])) {
             debugout::dump("WARNING: Failed to trim the silence from {$show_data['arrTracks'][1]['fileSource']}");
         }
+        debugUnlink(Configuration::getWorkingDir() . '/' . $show_data['arrTracks'][1]['fileSource']);
 
-        if ( ! concatenateTracks(Configuration::getWorkingDir() . '/run.wav', Configuration::getWorkingDir() . '/' . $show_data['arrTracks'][1]['fileSource'], Configuration::getWorkingDir() . '/runplustrack.wav')) {
+
```
Comment (JonTheNiceGuy) [inline CLI/showmaker.php]: Too many blank lines

File: CLI/showmaker.php [hunk 4]
```diff
@@ -316,7 +318,7 @@
             echo $arrTrack['strTrackName'] . ' by ' . $arrTrack['strArtistName'];
         }
         echo PHP_EOL;
-        
+
```
Comment (JonTheNiceGuy) [inline CLI/showmaker.php]: Oops, my bad :) Good spot!

File: CLI/showmaker.php [hunk 5]
```diff
@@ -333,7 +335,7 @@
                     'Hey there <BREAK LEVEL="MEDIUM" /> You are listening to a feed from %2$s and this is the %1$s <BREAK LEVEL="MEDIUM" /> In this show you will hear ten great tracks that we played over the past two weeks <BREAK LEVEL="MEDIUM" /> '
                 )
             ),
-            $show_data['strShowNameSpoken'], 
+            $show_data['strShowNameSpoken'],
```
Comment (JonTheNiceGuy) [inline CLI/showmaker.php]: OK, also good spot here!

File: CLI/showmaker.php [hunk 6]
```diff
@@ -687,25 +693,31 @@
...
 
-            $running_order = addEntryToJsonArray($running_order, getTrackLength(Configuration::getWorkingDir() . '/runplusbumper.wav'), $arrTrack['intTrackID']);
+            $run_length += getTrackLength(Configuration::getWorkingDir() . '/bumper.' . $intTrackID . '.wav');
+            $running_order = addEntryToJsonArray($running_order, $run_length, $arrTrack['intTrackID']);
 
             debugout::add("Downloading and merging audio file ($intTrackID)" . PHP_EOL);
             $track = downloadFile($arrTrack['localSource']);
             if ($track === false) {
-                debugUnlink(Configuration::getWorkingDir() . '/runplusbumper.wav');
+//                debugUnlink(Configuration::getWorkingDir() . '/runplusbumper.wav');
```
Comment (JonTheNiceGuy) [inline CLI/showmaker.php]: Should we remove this one?

File: CLI/library.php (hunk 1 above)
Comment (thelovebug) [reply CLI/library.php]: Will be in next commit

File: CLI/library.php (hunk 2 above)
Comment (thelovebug) [reply CLI/library.php]: Will be in next commit

File: CLI/showmaker.php (hunk 3 above)
Comment (thelovebug) [reply CLI/showmaker.php]: Will be in next commit

File: CLI/showmaker.php (hunk 6 above)
Comment (thelovebug) [reply CLI/showmaker.php]: Will be in next commit

**Pull request merged**